cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java

```diff
@@ -248,10 +260,54 @@ public static void main(String[] args) {
         cliMain.tryAutoconnect();
         cliMain.updateCLIPrompt(cliConfig.getClientConfig());
 
+        if (launchOptions.getScriptFile() != null) {
+
+          ConnectionConfig connectionConfig = clientConfig.getConnectionConfig();
+          URI baseURI = connectionConfig.getURI();
+          URI uri = baseURI.resolve("/" + connectionConfig.getNamespace());
+          String cliPrompt = "\ncdap (" + uri + ")> ";
+
+          BufferedReader br = null;
+          try {
+            String line;
+            br = new BufferedReader(new FileReader(launchOptions.getScriptFile()));
+            while ((line = br.readLine()) != null) {
+              String ln = line.trim();
+              if ("".equals(ln) || ln.startsWith("#")) {
+                continue;
+              }
+              if (ln.startsWith("break")) {
+                output.println(cliPrompt + ln);
+                break; // Break from script
+              }
+              output.println(cliPrompt + ln);
+              cli.execute(ln, output);
+            }
+            output.println("\n");
+          } catch (IOException e) {
+            output.println(e.getMessage());
+            usage();
+          } finally {
+            try {
+              if (br != null) {
+                br.close();
+              }
+            } catch (IOException e) {
+              output.println(e.getMessage());
+              usage();
+            }
+          }
+        }
         if (commandArgs.length == 0) {
+          // No commands following switches; just start shell
           cli.startInteractiveMode(output);
         } else {
-          cli.execute(Joiner.on(" ").join(commandArgs), output);
+          // Commands following switches; process them
+          String commandLine = Joiner.on(" ").join(commandArgs);
+          String[] commandLns = commandLine.split(";");
```

Comment:
I'm not sure that I follow this.  Is a ';' character used within the cdap-cli command space?